scripts/performance/Perf_runner.py

```diff
@@ -1,118 +1,374 @@
+              file=result_file)
+        print("\n Estimated transactions per second: " + str(txns_per_second),
+              file=result_file)
+
+    def find_lowest_and_fastest_transaction(self):
+        """
+        Find lowest and fastest transactions.
+        """
+        self.lowest = self.list_tester[0].lowest_txn
+        self.fastest = self.list_tester[0].fastest_txn
+        for tester in self.list_tester:
+            temp_lowest = tester.lowest_txn
+            temp_fastest = tester.fastest_txn
+            if self.lowest < temp_lowest:
+                self.lowest = temp_lowest
+            if self.fastest > temp_fastest:
+                self.fastest = temp_fastest
+
+    def find_start_and_finish_time(self):
+        """
+        Find the earliest time that a client is started and latest time that a
+        client is finished.
+        """
+        self.start_time = self.list_tester[0].start_time
+        self.finish_time = self.list_tester[0].finish_time
+
+        for tester in self.list_tester:
+            if self.start_time > tester.start_time:
+                self.start_time = tester.start_time
+
+            if self.finish_time < tester.finish_time:
+                self.finish_time = tester.finish_time
+
+    def start_tester_in_thread(self):
+        """
+        Create thread and start all the tester in list.
+        """
+        threads = list()
+        for _ in range(self.options.clients):
+            tester = self.create_tester()
+            self.list_tester.append(tester)
+            thread = threading.Thread(target=self.run_tester_in_thread,
+                                      kwargs={'tester': tester})
+            thread.daemon = True
+            thread.start()
+            threads.append(thread)
+
+        for thread in threads:
+            thread.join()
+
+    @staticmethod
+    def run_tester_in_thread(tester):
+        """
+        Execute testing function of tester.
+        """
+        loop = asyncio.new_event_loop()
+        utils.run_async_method(loop, tester.test)
+        loop.close()
+
+    def create_tester(self):
+        """
+        Create tester base mode "-a", "-t", "-g", "-l".
+
+        :return: tester
+        """
+        if self.options.adding:
+            return perf_add_requests.PerformanceTesterForAddingRequest(
+                self.options.info_dir, self.options.txns, self.options.kind,
+                thread_num=self.options.thread_num, log=self.options.log)
+
+        elif self.options.getting:
+            return perf_get_requests.PerformanceTesterGetSentRequestFromLedger(
+                self.options.info_dir, self.options.kind,
+                self.options.thread_num, log=self.options.log)
+
+        elif self.options.loading:
+            return perf_load.TesterSimulateLoad(
+                self.options.clients, self.options.txns,
+                self.options.time_out, self.options.log)
+
+        elif self.options.simulate_traffic:
```

Comment:
It looks like all scripts can't send requests in a real traffic mode. All of them are waiting for the result of the previous request before sending the next one.